packages/babel-generator/src/generators/typescript.ts

```diff
@@ -614,8 +614,28 @@ export function TSEnumDeclaration(this: Printer, node: t.TSEnumDeclaration) {
   this.print(id);
   this.space();
 
+  if (process.env.BABEL_8_BREAKING) {
+    // @ts-ignore(Babel 7 vs Babel 8) Babel 8 AST
+    this.print(node.body);
+  } else {
+    printBraced(this, node, () =>
+      this.printList(
+        // @ts-ignore(Babel 7 vs Babel 8) Babel 7 AST
+        node.members,
+        {
+          indent: true,
+          statement: true,
+          // TODO: Default to false for consistency with everything else
+          printTrailingSeparator: this.shouldPrintTrailingComma("}") ?? true,
+        },
+      ),
+    );
+  }
+}
+
+export function TSEnumBody(this: Printer, node: t.TSEnumBody) {
   printBraced(this, node, () =>
-    this.printList(members, {
+    this.printList(node.members, {
       indent: true,
       statement: true,
       // TODO: Default to false for consistency with everything else
```

Comment:
While we are changing this code, could you make this default to `process.env.BABEL_8_BREAKING ? false : true`? :)